### ISSUE TITLE
Exception: RENAME EXCHANGE is not supported raised using Refreshable Materialized View

### ISSUE THREAD

**Issue by kilik52** [ClickHouse/ClickHouse#61826]
**How to reproduce**
* Which ClickHouse server version to use

ClickHouse client version 24.2.2.71 (official build).

* Non-default settings, if any

`set allow_experimental_refreshable_materialized_view = 1`

* `CREATE TABLE` statements for all tables involved
* 
````sql
CREATE MATERIALIZED VIEW  mv
REFRESH EVERY 1 MINUTE
(
    `count` Float64
)
ENGINE = MergeTree
ORDER BY count
AS SELECT count() from a;
````

When I run 

````sql
select * from system.view_refreshes
````

it show there is an exception:

````
Code: 48. DB::Exception: RENAME EXCHANGE is not supported. (NOT_IMPLEMENTED), Stack trace (when copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000cf5af1b
1. DB::Exception::Exception<>(int, FormatStringHelperImpl<>) @ 0x00000000078cc583
2. DB::DatabaseAtomic::renameTable(std::shared_ptr<DB::Context const>, String const&, DB::IDatabase&, String const&, bool, bool) @ 0x0000000010e5cf08
3. DB::StorageMaterializedView::exchangeTargetTable(DB::StorageID, std::shared_ptr<DB::Context const>) @ 0x0000000012488b4f
4. DB::RefreshTask::refreshTask() @ 0x0000000012e8da2e
5. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::RefreshTask::create(DB::StorageMaterializedView const&, std::shared_ptr<DB::Context>, DB::ASTRefreshStrategy const&)::$_0, void ()>>(std::__function::__policy_storage const*) @ 0x0000000012e8fcff
6. DB::BackgroundSchedulePool::threadFunction() @ 0x0000000010a285cd
7. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x0000000010a29693
8. void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000d0003de
9. ? @ 0x00007f711bdf6609
10. ? @ 0x00007f711bd1b353
 (version 24.2.2.71 (official build))
````

**Comment by Algunenano** [comment]
What operative system are you using? By the error message it seems that atomic exchange is not supported. It's only currently implemented for "modern" Linux (3.15+) and OSX. 


cc @al13n321 Maybe we should check for availability of `supportsAtomicRename` during refreshable MV creation (but not in attach).

**Comment by enteve** [comment]
docker, alpine

clickhouse/clickhouse-server:24.3.1.2672-alpine
